cli/options.go

```diff
@@ -37,53 +37,84 @@ type ProjectOptions struct {
 	Name        string
 	WorkingDir  string
 	ConfigPaths []string
-	Environment []string
+	Environment map[string]string
 }
 
-// WithOsEnv imports environment variables from OS until those have been overridden by ProjectOptions.Environment
-func (o ProjectOptions) WithOsEnv() ProjectOptions {
-	env := getAsEqualsMap(o.Environment)
-	for k, v := range getAsEqualsMap(os.Environ()) {
-		if _, ok := env[k]; !ok {
-			env[k] = v
+type ProjectOptionsFn func(*ProjectOptions) error
+
+// NewProjectOptions creates ProjectOptions
+func NewProjectOptions(configs []string, opts ...ProjectOptionsFn) (*ProjectOptions, error) {
+	options := &ProjectOptions{
+		ConfigPaths: configs,
+		Environment: map[string]string{},
+	}
+	for _, o := range opts {
+		err := o(options)
+		if err != nil {
+			return nil, err
 		}
 	}
+	return options, nil
+}
 
-	return ProjectOptions{
-		Name:        o.Name,
-		WorkingDir:  o.WorkingDir,
-		ConfigPaths: o.ConfigPaths,
-		Environment: getAsStringList(env),
+// WithName defines ProjectOptions' name
+func WithName(name string) ProjectOptionsFn {
+	return func(o *ProjectOptions) error {
+		o.Name = name
```

Comment:
Should validation take place at this point, or is that done later?

Reply:
could be done here. So far we haven't clarified name constraints